src/System.Management.Automation/engine/remoting/common/RemoteSessionNamedPipe.cs

```diff
@@ -108,30 +110,43 @@ internal static class NamedPipeUtils
                 // The starttime is there to prevent another process easily guessing the pipe name
                 // and squatting on it.
                 // There is a limit of 104 characters in total including the temp path to the named pipe file
-                // on non-Windows systems, so we'll convert the starttime to hex and just take the first 8 characters.
+                // on non-Windows systems, so we'll convert the starttime to hex and just take 8 characters.
 #if UNIX
                 .Append(proc.StartTime.ToFileTime().ToString("X8").AsSpan(1, 8))
 #else
                 .Append(proc.StartTime.ToFileTime().ToString(CultureInfo.InvariantCulture))
 #endif
                 .Append('.')
-                .Append(proc.Id.ToString(CultureInfo.InvariantCulture))
-                .Append('.')
+                .Append(proc.Id.ToString(CultureInfo.InvariantCulture));
+
+            int uniqueLength = pipeNameBuilder.Length;
+            pipeNameBuilder.Append('.')
                 .Append(CleanAppDomainNameForPipeName(appDomainName))
                 .Append('.')
                 .Append(proc.ProcessName);
+
+            string pipeName = pipeNameBuilder.ToString();
 #if UNIX
-            int charsToTrim = pipeNameBuilder.Length - MaxNamedPipeNameSize;
+            // .NET Creates named pipes in the tmp path with a special prefix which needs to be included in the max
+            // length calculation.
+            string pipePrefix = Path.Combine(Path.GetTempPath(), "CoreFxPipe_");
+            int remainingPipeLength = Math.Max(MaxNamedPipeNameSize - pipePrefix.Length, 0);
+
+            int charsToTrim = pipeName.Length - remainingPipeLength;
             if (charsToTrim > 0)
             {
                 // TODO: In the case the pipe name is truncated, the user cannot connect to it using the cmdlet
```

Comment:
I don't understand this comment.  If a pipe name cannot be connected to, then what good is it?

Reply:
This comment is what was already there, I didn't really touch it. Essentially what it means is that `Enter-PSHostProcess -Id ...` will enumerate existing pipes on the system and validate the pipe name matches the start time + process name. In the case of a trimmed pipe server it will fail to find the pipe for that process id and say

> Enter-PSHostProcess: Cannot enter process ... because it has not loaded the PowerShell engine.

What does still work is if you do [NamedPipeConnectionInfo](https://docs.microsoft.com/en-us/dotnet/api/system.management.automation.runspaces.namedpipeconnectioninfo?view=powershellsdk-7.0.0) and manually open the Runspace with the .NET methods or you have a 3rd party client trying to connect to it.